src/openapi/streaming/subscription.js

```diff
@@ -763,31 +762,44 @@ Subscription.prototype.processSnapshot = function(response) {
 };
 
 /**
- * This assumes the subscription is dead and subscribes again. If unsubscribed or awaiting a unsubscribe, this is ignored.
- * It should be used in the case of errors, such as the subscription becoming orphaned and when the server asks us to reset a subscription.
+ * Reset happens when the server notices that a publisher is dead or when
+ * it misses some messages so it doesn't know who is dead (reset all)
+ * This may be called with a burst of messages. The intent is that we queue
+ * an operation to unsubscribe, wait for that to finish and then subscribe
+ * This waiting means that if we get further resets whilst unsubscribing, we
+ * can ignore them. It also ensures that we don't hit the subscription limit
+ * because the subscribe manages to get to the server before the unsubscribe.
  * @private
  */
 Subscription.prototype.reset = function() {
     switch (this.currentState) {
         case this.STATE_UNSUBSCRIBED:
         case this.STATE_UNSUBSCRIBE_REQUESTED:
-            // do not do anything if we are on our way to unsubscribed unless the next action would be to subscribe
-            if (this.queue.peekAction() & ACTION_SUBSCRIBE) {
-                break;
-            }
+            // do not do anything - even if the next action is to subscribe, we can go ahead and do that when the unsubscribe response comes back

```

Comment:
> even if the next action is to subscribe, we can go ahead and do that when the unsubscribe response comes back

How will scenario with next subscribe know that queue should be reset? Wasn't previous solution clearing the queue if we knew that next action was an subscribe and we are now missing this functionality?

Reply:
If we have requested a unsubscribe our job is done - the subscription is on its way to being unsubscribed. If the next action is subscribe- that’s ok because it hasn’t hit the server yet so it can’t possibly be included in what needs to be reset.
What we did before meant that when we get 10 reset all in a row, we started the subscription 10 times.